verifiers/internal/gha/slsaprovenance/v1.0/provenance.go

```diff
@@ -112,29 +113,41 @@ func (prov *ProvenanceV1) Subjects() ([]intoto.Subject, error) {
 }
 
 func (prov *ProvenanceV1) GetBranch() (string, error) {
-	// TODO(https://github.com/slsa-framework/slsa-verifier/issues/472): Add GetBranch() support.
-	sysParams, ok := prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
-	if !ok {
-		return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "system parameters type")
+	// Returns the branch from the source URI.
+	sourceURI, err := prov.SourceURI()
+	if err != nil {
+		return "", err
+	}
+
+	parts := strings.Split(sourceURI, "@")
+	if len(parts) > 1 && strings.HasPrefix(parts[1], "refs/heads") {
+		return parts[1], nil
 	}
 
-	return slsaprovenance.GetBranch(sysParams, prov.predicateType)
+	return "", nil
 }
 
 func (prov *ProvenanceV1) GetTag() (string, error) {
-	sysParams, ok := prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
```

Comment:
do we need a TODO here for https://github.com/slsa-framework/slsa-verifier/issues/472?

Reply:
#472 is closed and I wasn't sure what it was referring to in this case. I removed the TODO above because it was closed.

Do you think we should re-open it?

Reply:
iiuc, this PR removes support for branch / tag for v1?

Reply:
branch / tag verification won't work for v1 if the trw specifies the a different sha than the trigger sha but it should still work otherwise. At least that's my intention.